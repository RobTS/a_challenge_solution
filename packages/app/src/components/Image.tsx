import FastImage, { ImageStyle } from 'react-native-fast-image';
import { StyleProp } from 'react-native';
import React from 'react';

type ImageProps = {
  url: string;
  style?: StyleProp<ImageStyle>;
};

export const Image: React.FC<ImageProps> = (props: ImageProps) => (
  <FastImage
    style={props.style}
    source={{
      uri: props.url,
      priority: FastImage.priority.normal,
    }}
    resizeMode={FastImage.resizeMode.contain}
  />
);
