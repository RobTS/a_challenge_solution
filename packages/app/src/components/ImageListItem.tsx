import { StyleSheet, View } from 'react-native';
import React from 'react';
import { Image } from './Image';

type ImageProps = {
  url: string;
};

export const ImageListItem: React.FC<ImageProps> = (props: ImageProps) => {
  return (
    <View style={styles.imageListItem}>
      <Image url={props.url} style={styles.image} />
    </View>
  );
};

const styles = StyleSheet.create({
  imageListItem: {
    margin: 10,
  },
  image: { flex: 1 },
});
