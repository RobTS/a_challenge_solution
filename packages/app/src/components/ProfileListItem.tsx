import { StyleSheet, Text, View } from 'react-native';
import React from 'react';
import { Image } from './Image';
import { Profile } from '../state/entities';

export const ProfileListItem: React.FC<Profile> = (props: Profile) => {
  return (
    <View style={styles.imageListItem}>
      <Image url={props.avatar} style={styles.image} />
      <Text>{props.name}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  imageListItem: {
    margin: 10,
  },
  image: { flex: 1 },
});
