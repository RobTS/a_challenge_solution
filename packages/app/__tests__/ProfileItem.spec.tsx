import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { ProfileListItem } from '../src/components/ProfileListItem';

describe('The profile item', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ProfileListItem
          even={true}
          name={'Peter Testman'}
          id={'2'}
          avatar={'https://i.pravatar.cc/150?img=2'}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
