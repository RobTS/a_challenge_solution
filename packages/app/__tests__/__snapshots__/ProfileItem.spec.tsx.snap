// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The profile item renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "position": "relative",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#000000",
          "flex": 1,
          "height": 421.875,
          "width": 750,
        },
      ]
    }
  >
    <FastImageView
      resizeMode="cover"
      source={
        Object {
          "priority": "normal",
          "uri": "https://i.pravatar.cc/150?img=2",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    numberOfLines={1}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "bottom": 30,
        "fontWeight": "bold",
        "left": 0,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
        "position": "absolute",
      }
    }
  >
    Peter Testman
  </Text>
</View>
`;
